### PR TITLE
fix:  Changes in web3modal are reflected in the etherium provider package.

### PR DESCRIPTION
**Pull request by siosio34**
# Description

I know that wagmi wraps etherium-provider when connecting walletconnect.

So I referred to https://docs.walletconnect.com/2.0/web/providers/ethereum to find qrmodeOptions that can be customized in the ethereum provider.

Then, while the https://docs.walletconnect.com/2.0/web/providers/ethereum document uses the latest version of web3modal's qrModalOptions,  I noticed that the Ethereum Provider package was mapped to the contents of the previous version of web3modal.

So I proceeded to upgrade the dependency of the Ethereum Provider and adjust the type so that I can use the latest version of the web3modal option.

Resolves # ([browse](https://github.com/WalletConnect/walletconnect-monorepo/issues) or [create](https://github.com/WalletConnect/walletconnect-monorepo/issues/new/choose))

## Due Diligence

- [] Breaking change
- [] Requires a documentation update
  - [] Related docs issue/PR (if docs are not included in this PR):
- [] Requires a e2e/integration test update

### CODE REVIEW

**Comment by ganchoradkov** [comment]
Hey @siosio34, thanks for the PR. Is it possible to resolve the `package-lock.json` conflict by merging `v2.0` and to exclude the `yarn.lock` from it?

File: providers/ethereum-provider/src/EthereumProvider.ts [hunk 1]
```diff
@@ -18,13 +18,17 @@ export type QrModalOptions = Pick<
...
   | "enableExplorer"
-  | "explorerAllowList"
-  | "explorerDenyList"
+  | "explorerRecommendedWalletIds"
+  | "explorerExcludedWalletIds"
   | "mobileWallets"
   | "privacyPolicyUrl"
   | "termsOfServiceUrl"
   | "tokenImages"
+  | "tokenContracts"
```
Comment (xzilja) [inline providers/ethereum-provider/src/EthereumProvider.ts]: `tokenContracts`, `standaloneChains`, `enableNetworkView`, `enableAccountView` options are not available here (standalone modal).

**Review by xzilja — CHANGES_REQUESTED** [review]

File: providers/ethereum-provider/src/EthereumProvider.ts (hunk 1 above)
Comment (siosio34) [reply providers/ethereum-provider/src/EthereumProvider.ts]: f0e702a55c4c21f2d7bd32df61ef10c044bb66ee 


oh i didn't know thank you for telling me. Corrected this. Can you check when you have time?

**Comment by siosio34** [comment]
@ganchoradkov

The work you mentioned was done in the commit below. Please check when you have time.

[f0e702a](https://github.com/WalletConnect/walletconnect-monorepo/pull/2289/commits/f0e702a55c4c21f2d7bd32df61ef10c044bb66ee)

**Review by ganchoradkov — APPROVED** [review]
Awesome!

**Pull request merged**